Consul/Agent.cs

```diff
@@ -773,6 +778,12 @@ internal LogStream(Stream s)
             }
 
             public void Dispose()
+            {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+
+            public virtual void Dispose(bool disposing)
             {
```

Comment:
the `if (disposing)` is necessary for the dispose pattern to be correct.

Reply:
oh you're right